<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="panda_pba_robots">

    <!-- World fixed link -->
    <link name="world"/>
    
    <!-- Import UR20 Robot - directly include without macro call -->
    <xacro:include filename="$(find sirgas_description)/urdf/panda.urdf.xacro" />

    <!-- Import PBA Robot - directly include without macro call -->
    <xacro:include filename="$(find sirgas_description)/urdf/pba2.urdf.xacro" />

    <!-- Import PBA Robot - directly include without macro call -->
    <xacro:include filename="$(find sirgas_description)/urdf/camera2.urdf.xacro" />

    <ros2_control name="combined_controller" type="system">  
        <hardware>
            <plugin>gz_ros2_control/GazeboSimSystem</plugin>
        </hardware>

        <!-- Panda Robot -->
        <joint name="panda_joint1">
            <command_interface name="position"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="panda_joint2">
            <command_interface name="position"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="panda_joint3">
            <command_interface name="position"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="panda_joint4">
            <command_interface name="position"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="panda_joint5">
            <command_interface name="position"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="panda_joint6">
            <command_interface name="position"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>
        <joint name="panda_joint7">
            <command_interface name="position"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>

        <joint name="panda_finger_joint1">
            <command_interface name="position"/> <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="panda_finger_joint2">
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        
        <!-- PBA Robot -->
        <joint name="pb_starter_joint">
            <command_interface name="velocity">
                <param name="min">-10</param>
                <param name="max">10</param>
            </command_interface>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="pb_first_joint">
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="pb_second_joint">
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
        <joint name="pb_third_joint">
            <state_interface name="position"/>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
    </ros2_control>

    <gazebo>
        <plugin filename="libgz_ros2_control-system.so" name="gz_ros2_control::GazeboSimROS2ControlPlugin">
            <parameters>$(find sirgas_description)/config/test.yaml</parameters>
            <controller_manager_name>combined_controller_manager</controller_manager_name>
        </plugin>
    </gazebo>

</robot>